app/grandchallenge/cases/image_builders/tiff.py

```diff
@@ -0,0 +1,138 @@
+"""
+Image builder for TIFF files.
+"""
+
+from pathlib import Path
+from django.core.files import File
+
+import tifffile as tiff_lib
+from grandchallenge.cases.image_builders import ImageBuilderResult
+from grandchallenge.cases.models import Image, ImageFile
+
+
+def image_builder_tiff(path: Path) -> ImageBuilderResult:
+    new_images = []
+    new_image_files = []
+    consumed_files = set()
+    invalid_file_errors = {}
+
+    for file in path.iterdir():
+        valid, message = validate_tiff(file.absolute())
+
+        if not valid:
+            invalid_file_errors[file.name] = message
+        else:
+            new_images.append(
+                create_tiff_image_entry(file.name, file.absolute())
```

Comment:
Why pass both arguments to this function when the 2nd contains all of the information you need?